### PR TITLE
Similarity solution viscosity analysis

### PR DESCRIPTION
**Pull request by matthiasto**
Split com1DFA simi sol run from the analysis run
Improve plot

Previous PR https://github.com/avaframe/AvaFrame/pull/630

### CODE REVIEW

File: avaframe/out3Plot/outAna1Plots.py [hunk 1]
```diff
@@ -29,7 +29,9 @@ def showSaveTimeSteps(cfgMain, cfgSimi, particlesList, fieldsList, solSimi, Tsav
     if cfgSimi.getboolean('interaction'):
         value = input("give time step to plot (float in s):\n")
     else:
-        value = cfgSimi.getfloat('tSave')
```
Comment (awirb) [inline avaframe/out3Plot/outAna1Plots.py:32]: in line 24 remove commented line?

File: avaframe/out3Plot/outAna1Plots.py [hunk 2]
```diff
@@ -233,152 +259,329 @@ def plotErrorTime(time, hErrorL2Array, hErrorLMaxArray, vhErrorL2Array, vhErrorL
...
-        p1U = p[0]
-        p0U = np.exp(p[1])
-        ax1.plot(nPart, p0H*nPart**p1H, 'r')
-        ax2.plot(nPart, p0U*nPart**p1U, 'g')
-        log.info('power law fit sphKernelRadius = %.2f m: hErrorL2 = %.1f * nPart^{%.2f}' % (sphKernelRadius, p0H, p1H))
-        log.info('power law fit sphKernelRadius = %.2f m: vhErrorL2 = %.1f * nPart^{%.2f}' % (sphKernelRadius, p0U, p1U))
-    ax1.set_yscale('log')
-    ax2.set_yscale('log')
-    ax1.set_xscale('log')
+    # for the scater plot (new filtering)
```
Comment (awirb) [inline avaframe/out3Plot/outAna1Plots.py:412]: scatter

File: avaframe/runScripts/runAnalyzeSimilaritySol.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+log.info('Current avalanche: %s', avalancheDir)
+
+# Load configuration for similarity solution test
+simiSolCfg = pathlib.Path(avalancheDir, 'Inputs', 'simiSol_com1DFACfg.ini')
+
+# create output directory for test result plots
+outDirTest = pathlib.Path(avalancheDir, 'Outputs', 'ana1Tests')
+fU.makeADir(outDirTest)
+
+cfg = cfgUtils.getModuleConfig(com1DFA, simiSolCfg)
```
Comment (awirb) [inline avaframe/runScripts/runAnalyzeSimilaritySol.py:49]: move this line below line 43

File: avaframe/runScripts/runAnalyzeSimilaritySol.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+# Define release thickness distribution
+demFile = gI.getDEMPath(avalancheDir)
+simDF, _ = cfgUtils.readAllConfigurationInfo(avalancheDir)
+
+# compute the similartiy solution (this corresponds to our reference)
+log.info('Computing similarity solution')
+solSimi = simiSolTest.mainSimilaritySol(simiSolCfg)
+
+# if the analysis already exists and you only want to replot uncoment this
```
Comment (awirb) [inline avaframe/runScripts/runAnalyzeSimilaritySol.py:59]: uncomment

File: avaframe/runScripts/runAnalyzeSimilaritySol.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+# pathToResults = pathlib.Path(avalancheDir, 'Outputs', 'ana1Tests', 'results10.p')
```
Comment (awirb) [inline avaframe/runScripts/runAnalyzeSimilaritySol.py:60]: is it always for 10s? or should this name be build or read?

File: avaframe/runScripts/runAnalyzeSimilaritySol.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+#     simDF = pd.read_pickle(pathToResults)
+simDF = simiSolTest.postProcessSimiSol(avalancheDir, cfgMain, cfg['SIMISOL'], simDF, solSimi, outDirTest)
+
+# select the simulations you want to plot
+# filter on viscosity parameter
+# simDF = simDF[simDF['subgridMixingFactor'].isin([10])]
+# filter on time stepping parameter
+# simDF = simDF[simDF['cMax'].isin([0.01])]
+# filter on nPPK0
+# simDF = simDF[simDF['nPPK0']==15]
```
Comment (awirb) [inline avaframe/runScripts/runAnalyzeSimilaritySol.py:71]: could this become part of the ini file to filter according to certain criteria?

**Review by awirb — CHANGES_REQUESTED** [review]

File: avaframe/runScripts/runAnalyzeSimilaritySol.py (hunk 6 above)
Comment (matthiasto) [reply avaframe/runScripts/runAnalyzeSimilaritySol.py:71]: It is easier to filter multiple parameters at a time in the script as in the ini file

File: avaframe/runScripts/runAnalyzeSimilaritySol.py (hunk 5 above)
Comment (matthiasto) [reply avaframe/runScripts/runAnalyzeSimilaritySol.py:60]: No this can change. You can decide to analyze the results for other times

**Review by fso42 — APPROVED** [review]

File: avaframe/com1DFA/timeDiscretizations.py [hunk 7]
```diff
@@ -45,3 +45,34 @@ def getcflTimeStep(particles, dem, cfg):
...
+        corresponding time step
+    """
+    # get cell size
+    cszDEM = dem['header']['cellsize']
+    cszNeighbourGrid = dem['headerNeighbourGrid']['cellsize']
+    # use the smallest of those two values
+    csz = min(cszDEM, cszNeighbourGrid)
+
+    # courant number
+    cMax = float(cfg['cMax'])
```
[the remark this answers is not in the document]
Comment (matthiasto) [reply avaframe/com1DFA/timeDiscretizations.py:72]: Not realy in the sense of CFL where you have dt = CFL * dx/v, but it serves a similar purpose to limit the max dt.
I am open to other naming

**Pull request merged**